server/chat-plugins/chat-monitor.ts

```diff
@@ -20,7 +21,30 @@ interface Monitor {
 	monitor?: MonitorHandler;
 }
 
-const MONITOR_FILE = 'config/chat-plugins/chat-monitor.tsv';
+interface EvaderMonitor {
+	/** The name of the monitor */
+	name: string;
+	/** Whether or not autoconfirmed users bypass this monitor */
+	autoconfirmedBypass?: boolean;
+	/** This function will be invoked when the monitor is triggered. */
+	action: (user: User, monitor: EvaderMonitor) => void;
+	/** If specified, the user's IP must be in one of these ranges to activate the monitor. */
+	ips?: AddressRange[];
+	/** If specified, the user's host must include one of the strings in this array.  */
+	hostContents?: string[];
+	/** If specified, the user's ID must include one of the strings in this array.  */
+	idContents?: string[];
+	/**
+	 * Each key in fields represents an attribute of a User object.
+	 * The attribute's value must be in the provided array for the monitor to activate.
+	 */
+	fields?: {[k: string]: any[]};
+	/** Custom function that must return true if specified to activate the monitor. */
+	checker?: (user: User) => boolean;
+}
+
+const EVADER_MONITOR_FILE = '../../config/monitors.js';
+const CHAT_FILTER_FILE = 'config/chat-plugins/chat-monitor.tsv';
```

Comment:
I take it you renamed to `CHAT_FILTER_FILE` to reduce confusion with the evader monitor. Is there a plan to rename the tsv, too? Otherwise, it might make sense to stay consistent with the established naming.

Reply:
I can rename the tsv too; it'd just make things a bit more complex since someone would need to rename it on the server before hotpatching.